app/os_macos.go

```diff
@@ -539,6 +546,11 @@ func gio_onMouse(view, evt C.CFTypeRef, cdir C.int, cbtns C.NSUInteger, x, y, dx
 	if cbtns&(1<<2) != 0 {
 		btns |= pointer.ButtonTertiary
 	}
+
+	if typ == pointer.Press {
+		C.setFocus(w.window, w.view)
```

Comment:
Can you track the focused state and only call setFocus if not focused? Similar to your Windows PR #86.

Reply:
It's a little bit different than Windows, I implemented the `becomeFirstResponder` (https://developer.apple.com/documentation/uikit/uiresponder/1621113-becomefirstresponder).

Since it was introduced, I also include `key.FocusEvent` here (the system.FocusEvent should not be triggered in that case), that may happens when another window (inside the Gio window) gets the "Focus", the `windowDidBecomeKey` isn't triggered in those cases.